lib/ae_mdw/db/mem_store.ex

```diff
@@ -0,0 +1,116 @@
+defmodule AeMdw.Db.MemStore do
+  @moduledoc """
+  Store implementation with operations reading/writing in-memory.
+
+  Uses a GbTree implementation for fast access to the keys, plus, being able to
+  iterate over them both forwards and backwards.
+
+  Fallbacks to a different Store implementation that can be configured when
+  building this store.
+  """
+
+  alias AeMdw.Database
+  alias AeMdw.Db.Store
+  alias AeMdw.Util.GbTree
+
+  @derive AeMdw.Db.Store
+  defstruct [:fallback_store, :tables]
+
+  @typep key() :: Database.key()
+  @typep record() :: Database.record()
+  @typep table() :: Database.table()
+  @opaque t() :: %__MODULE__{
+            fallback_store: Store.t(),
```

Comment:
This module is very neat and the code clear, maybe we should only analyse if time allows whether it would make more sense for the `State` to do it accessing different stores.

Reply:
From one conceptual point of view a `State` would know where to look for the info (when it doesn't exist in memory) but the way it is is elegant because keeps the `MemStore` fully working as the behaviour it implements.